zcash_client_backend/src/data_api/wallet.rs

```diff
@@ -433,7 +489,7 @@ pub fn propose_standard_transfer_to_address<DbT, ParamsT, CommitmentTreeErrT>(
     wallet_db: &mut DbT,
     params: &ParamsT,
     fee_rule: StandardFeeRule,
-    spend_from_account: AccountId,
+    spend_from_account: <DbT as WalletRead>::AccountId,
```

Comment:
The trait bounds are inconsistent as to whether `WalletRead::AccountId` or `InputSource::AccountId` is authoritative. It currently doesn't matter because we have a bound that requires the two be the same type, but for consistency (and to avoid trait bound resolution hell later on) we should not mix and match when both trait bounds are present.

Reply:
I basically consider `InputSource::AccountId` as authoritative; it's the more limited trait. I need to remember why `propose_standard_transfer_to_address` still needs `WalletRead`; I've been meaning to prune down the capabilities demanded by these methods but haven't gotten there yet.

Reply:
Ah, right, it's just `get_target_and_anchor_heights`. So you're correct that this ought to be referring to the `InputSource` associated type.